### PR TITLE
Restructure Metric Logging and Add support for W&B

### PR DESCRIPTION
**Pull request by manangoel99**
Based on #5842 

This PR redesigns the logging system in PaddleOCR without changing any of the console logging. The proposed change is to have a `BaseLogger` class from which new loggers can inherit. This includes the VisualDL logger, the new Weights & Biases Logger and any new custom loggers that might be added later. 
Using the W&B Logger would be really easy for users. The only requirement is to set `use_wandb: True` in the `Global` config.yml file. To add any other `wandb` related arguments like project name, entity name and so on, the users can add the `wandb` key.
For example:
```
Global:
    use_wandb: True

wandb:
    project: PaddleOCR
    entity: MyTeam
```
This leads to all the training metrics being logged to W&B. The dashboard:
Screenshot 2022-03-30 at 8 57 00 PM
Screenshot 2022-03-30 at 8 57 13 PM

The models at every model saving step and evaluation step are also logged to the W&B run with the appropriate tags and metadata.
Screenshot 2022-03-30 at 8 59 08 PM

An example dashboard is available [here](https://wandb.ai/manan-goel/PaddleOCR/runs/27e7nb40)

### CODE REVIEW

**Comment by manangoel99** [comment]
@WenmuZhou @GreatV I'm not sure what would be a good place to add how to see metrics logged to W&B. Could you point me to the right place in the documentation?

**Comment by GreatV** [comment]
> @WenmuZhou @GreatV I'm not sure what would be a good place to add how to see metrics logged to W&B. Could you point me to the right place in the documentation?

I think you could add a separate README to demonstrate W&B related configuration.

**Comment by WenmuZhou** [comment]
You can add a log.md in `docs/doc_ch` and a log_en.md in `docs/doc_en`

**Comment by WenmuZhou** [comment]
Further, we now three logs, txt_logger, vdl_logger, wandblogger, these three loggers can be unified to make a call and perform their respective methods? for example
```bash
class Loggers:
    def __init__(self, log_params:dict) -> None:
        self.logs = []
        for log_param in log_params:
            self.logs.append(init_log(log_param))

    def info(metrics: dict):
        for log in self.logs:
            log.info(metrics)
```

**Comment by manangoel99** [comment]
@WenmuZhou Thanks for the suggestions and I've made the suggested changes. I would love some help with the chinese documentation. I have added the english version for now.

**Review by WenmuZhou — APPROVED** [review]

**Comment by WenmuZhou** [comment]
@GreatV can you give a ch doc?

**Comment by WenmuZhou** [comment]
Do I need to specify something like apikey when using wandb？

**Comment by GreatV** [comment]
> @GreatV can you give a ch doc?

No problem, I will provide a Chinese version of the document.

File: doc/doc_en/logging_en.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+To add more arguments to the `WandbLogger` listed [here](./config_en.md) add the header `wandb` to the yaml file and add the arguments under it - 
+
+```
+wandb:
+    project: my_project
+    entity: my_team
+```
+
+This will automatically log all the training and evaluation metrics to the W&B dashboard along with models at every model saving step and evaluation step are with the appropriate tags and metadata.
```
Comment (littletomatodonkey) [inline doc/doc_en/logging_en.md:46]: Thanks for your contribution! Could you please provide more detailed information on how to use the wandb tool for new users? For example, given a configuration file, how can we visualize using wandb step by step.

Comment (manangoel99) [reply doc/doc_en/logging_en.md]: Thanks for the suggestion! Please have a look and let me know if the changed version is better

**Comment by manangoel99** [comment]
@littletomatodonkey @WenmuZhou would love to hear your thoughts on the PR

**Review by littletomatodonkey — APPROVED** [review]

**Review by WenmuZhou — APPROVED** [review]

**Pull request merged**